AppCreationScripts/Configure.ps1

```diff
@@ -0,0 +1,187 @@
+    # you'll need to sign-in with creds enabling your to create apps in the tenant)
+    if (!$Credential -and $TenantId)
+    {
+        $creds = Connect-AzureAD -TenantId $tenantId
+    }
+    else
+    {
+        if (!$TenantId)
+        {
+            $creds = Connect-AzureAD -Credential $Credential
+        }
+        else
+        {
+            $creds = Connect-AzureAD -TenantId $tenantId -Credential $Credential
+        }
+    }
+
+    if (!$tenantId)
+    {
+        $tenantId = $creds.Tenant.Id
+    }
+    $tenant = Get-AzureADTenantDetail
+    $tenantName =  ($tenant.VerifiedDomains | Where { $_._Default -eq $True }).Name
+
+   # Create the app AAD application
+   Write-Host "Creating the AAD appplication (Webapp-Openidconnect)"
+   # Get a 2 years application key for the app Application
+   $pw = ComputePassword
+   $fromDate = [DateTime]::Now;
+   $key = CreateAppKey -fromDate $fromDate -durationInYears 2 -pw $pw
+   $appAppKey = $pw
+   $appAadApplication = New-AzureADApplication -DisplayName "Webapp-Openidconnect" `
+                                               -HomePage "https://localhost:8000/" `
+											   -ReplyUrls "http://localhost:8080/adal4jsample/secure/aad" `
+                                               -IdentifierUris "https://$tenantName/Webapp-Openidconnect" `
+                                               -PasswordCredentials $key `
+                                               -PublicClient $False
+
+
+   $currentAppId = $appAadApplication.AppId
+   $appServicePrincipal = New-AzureADServicePrincipal -AppId $currentAppId -Tags {WindowsAzureActiveDirectoryIntegratedApp}
+   Write-Host "Done."
+
+   # URL of the AAD application in the Azure portal
+   $appPortalUrl = "https://portal.azure.com/#@"+$tenantName+"/blade/Microsoft_AAD_IAM/ApplicationBlade/appId/"+$appAadApplication.AppId+"/objectId/"+$appAadApplication.ObjectId
+   Add-Content -Value "<tr><td>app</td><td>$currentAppId</td><td><a href='$appPortalUrl'>Webapp-Openidconnect</a></td></tr>" -Path createdApps.html
+
+   $requiredResourcesAccess = New-Object System.Collections.Generic.List[Microsoft.Open.AzureAD.Model.RequiredResourceAccess]
+   # Add Required Resources Access (from 'app' to 'Microsoft Graph')
+   Write-Host "Getting access from 'app' to 'Microsoft Graph'"
+   $requiredPermissions = GetRequiredPermissions -applicationDisplayName "Microsoft Graph" `
+                                                 -requiredDelegatedPermissions "Directory.Read.All";
+   $requiredResourcesAccess.Add($requiredPermissions)
+   Set-AzureADApplication -ObjectId $appAadApplication.ObjectId -RequiredResourceAccess $requiredResourcesAccess
+   Write-Host "Granted."
+   Write-Host ""
+   Write-Host "IMPORTANT: Think of completing the following manual step(s) in the Azure portal":
+   Write-Host "- For 'app'"
+   Write-Host "  - Navigate to '$appPortalUrl'"
+   Write-Host "  - Open src/main/webapp/WEB-INF/web.xml and replace the following values:"
```

Comment:
Great work around!

Reply:
@sangonzal Why does the configure script not populate these values into web.xml? Is this something to do with the xml file format? This work around was helpful when I tried it but why do we need a work around? Sorry I don't understand this.

Reply:
Work around was needed because the app creation script tool cannot write to nested xml files at the moment. That is functionality that Jean-Marc hopes to add at some point.